build/patches/Add-webrtc-site-setting.patch

```diff
@@ -0,0 +1,381 @@
+@@ -0,0 +1,85 @@
++/*
++    This file is part of Bromite.
++
++    Bromite is free software: you can redistribute it and/or modify
++    it under the terms of the GNU General Public License as published by
++    the Free Software Foundation, either version 3 of the License, or
++    (at your option) any later version.
++
++    Bromite is distributed in the hope that it will be useful,
++    but WITHOUT ANY WARRANTY; without even the implied warranty of
++    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++    GNU General Public License for more details.
++
++    You should have received a copy of the GNU General Public License
++    along with Bromite. If not, see <https://www.gnu.org/licenses/>.
++*/
++
++package org.chromium.components.browser_ui.site_settings;
++
++import org.chromium.components.browser_ui.site_settings.ContentSettingsResources;
++import org.chromium.components.browser_ui.site_settings.SiteSettingsCategory;
++import org.chromium.components.content_settings.ContentSettingValues;
++import org.chromium.components.content_settings.ContentSettingsType;
++import org.chromium.content_public.browser.BrowserContextHandle;
++
++import androidx.annotation.Nullable;
++import androidx.preference.Preference;
++import androidx.preference.PreferenceScreen;
++
++import java.util.ArrayList;
++
++public class BromiteWebRTCContentSetting extends BromiteCustomContentSetting {
++    public BromiteWebRTCContentSetting() {
++        super(/*contentSettingsType*/ ContentSettingsType.WEBRTC,
++              /*siteSettingsCategory*/ SiteSettingsCategory.Type.WEBRTC,
++              /*initialDefaultValue*/ ContentSettingValues.BLOCK,
++              /*defaultDisabledValue*/ ContentSettingValues.BLOCK,
++              /*allowException*/ true,
++              /*preferenceKey*/ "webrtc",
++              /*profilePrefKey*/ "webrtc");
++    }
++
++    @Override
++    public ContentSettingsResources.ResourceItem getResourceItem() {
++        return new ContentSettingsResources.ResourceItem(
++            /*icon*/ R.drawable.web_asset,
++            /*title*/ R.string.webrtc_permission_title,
++            /*defaultEnabledValue*/ getInitialDefaultValue(),
++            /*defaultDisabledValue*/ getDefaultDisabledValue(),
++            /*enabledSummary*/ R.string.website_settings_category_webrtc_enabled,
++            /*disabledSummary*/ R.string.website_settings_category_webrtc_disabled);
++    }
++
++    @Override
++    public int getCategorySummary(@Nullable @ContentSettingValues int value) {
++        switch (value) {
++            case ContentSettingValues.ALLOW:
++                return R.string.website_settings_category_webrtc_enabled;
++            case ContentSettingValues.BLOCK:
++                return R.string.website_settings_category_webrtc_disabled;
++            default:
++                return 0;
```

Comment:
When is this used?

Reply:
here
https://github.com/bromite/bromite/blob/master/build/patches/Content-settings-infrastructure.patch#L641

![image](https://user-images.githubusercontent.com/29201891/168413930-cfc40540-4569-4a01-958d-eb177c78b9bf.png)

it is the value that appears there, when opening the settings panel with the summary of all contentsettings, and it can be a different string from the other used by the interface as an option for the setting.

Reply:
Please add a comment since this `0` is a magic constant.

Reply:
> Please add a comment since this 0 is a magic constant.

ok, the comment is "with zero a java exception is thrown". I add it

Reply:
I will add the comment.